source/common/upstream/cluster_manager_impl.cc

```diff
@@ -1274,6 +1275,10 @@ void ClusterManagerImpl::postThreadLocalClusterUpdate(ClusterManagerCluster& cm_
             cluster_manager->thread_local_clusters_.size());
       }
 
+      if (cluster_manager->thread_local_clusters_[info->name()]) {
+        cluster_manager->thread_local_clusters_[info->name()]->setDropOverload(
+            cm_cluster.cluster().dropOverload());
```

Comment:
I'm a bit worried here that this value is being read by the worker threads while it can be written by the main thread. Can you add a comment explaining why this is safe.

Reply:
+1

Reply:
added

Reply:
Good catch. I passed cm_clusters as reference which left the door open for the race.  
Done!